hack/tools/install_kubebuilder.sh

```diff
@@ -2,14 +2,7 @@
 
 [[ -f bin/kubebuilder ]] && exit 0
 
-version=2.2.0
-arch=amd64
-
 mkdir -p ./bin
-curl -L -O "https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${version}/kubebuilder_${version}_linux_${arch}.tar.gz"
-
-tar -zxvf kubebuilder_${version}_linux_${arch}.tar.gz
-mv kubebuilder_${version}_linux_${arch}/bin/* bin
-
-rm kubebuilder_${version}_linux_${arch}.tar.gz
-rm -r kubebuilder_${version}_linux_${arch}
+cd ./bin || exit
+curl -L -o kubebuilder https://go.kubebuilder.io/dl/latest/"$(go env GOOS)"/"$(go env GOARCH)"
```

Comment:
I am thinking isn't it wise to fix the version? If you download it latest always, theres a chance CI and user would have separate versions.

Reply:
I agree that pinning a version would be much better and safer. However, I could find a URl to download only kubebuilder binary with a specific version. For example, if you replace latest with some semantic version, it won't work.

Reply:
what if we download it from github releases ? https://github.com/kubernetes-sigs/kubebuilder/releases/download/v3.2.0/kubebuilder_linux_arm64

Reply:
ow, I missed that. This should work indeed! Thank you. Which version shall we pin @kashifest ?

Reply:
We havent uplifted kubebuilder to v3 for capm3. I am a bit unsure how to proceed in this. May be the latest version which support v2?

Reply:
sure. [v2.3.2](https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v2.3.2) seems to be the last v2 release.

Reply:
@kashifest Actually it is only from release v3.0.0 when only kubebuilder binary is shipped. So, if we use v2 based releases, then we will be downloading other binaries too as part of the same tarball